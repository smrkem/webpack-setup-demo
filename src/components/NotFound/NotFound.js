import React from 'react'

const NotFound = () => (
  <div className="container">
    <h1 className="heading">Oops - couldn't find what you were looking for :(</h1>
  </div>
)

export default NotFound
