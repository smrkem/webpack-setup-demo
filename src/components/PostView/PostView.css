.post-view h2 {
  font-size: 3rem;
  color: #53384c;
}

.post-view .post-content {
  font-size: 1.8rem;
  line-height: 1.7;
}
