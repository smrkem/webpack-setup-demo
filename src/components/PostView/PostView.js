import React from 'react'
import posts from '../../posts.json'
import NotFound from '../NotFound/NotFound'
import './PostView.css'

const PostView = ({match}) => {
  let slug = match.params.slug
  let post = posts.filter(post => post.slug === slug)[0]
  if (typeof post == 'undefined') {
    return <NotFound />
  }

  return (
    <div className="container post-view">
      <h2>{ post.title }</h2>
      <div className="post-content">
        { post.content.split('\n\n').map((paragraph, i) => (
          <p key={i}>{ paragraph }</p>
        ))}
      </div>
    </div>
  )
}

export default PostView
