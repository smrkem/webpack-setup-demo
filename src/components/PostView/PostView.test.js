import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import PostView from './PostView'
import NotFound from '../NotFound/NotFound'

test('PostView should render ok', () => {
  const match = {
    params: {
      slug: "sample-post-1"
    }
  }
  const component = shallow(<PostView match={match} />)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('PostView should return NotFound with invalid slug', () => {
  const match = {
    params: {
      slug: "invalid-slug"
    }
  }
  const component = mount(<PostView match={match} />)
  expect(component.find(NotFound).length).toBe(1)
})
