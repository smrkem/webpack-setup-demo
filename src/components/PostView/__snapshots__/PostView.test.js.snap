// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostView should render ok 1`] = `
<div
  className="container post-view"
>
  <h2>
    Sample Post 1
  </h2>
  <div
    className="post-content"
  >
    <p>
      Hammock locavore bespoke williamsburg. Organic adaptogen offal single-origin coffee tattooed prism tumeric. Listicle fixie polaroid tbh la croix man braid celiac. Vice flannel health goth chia, organic shaman truffaut kinfolk narwhal quinoa echo park messenger bag. Occupy vape disrupt, try-hard scenester food truck whatever aesthetic vexillologist. Kogi dreamcatcher authentic, selfies hella vaporware keffiyeh asymmetrical. Gluten-free hell of air plant polaroid, glossier bicycle rights salvia master cleanse jianbing post-ironic portland chillwave occupy. Chicharrones schlitz hell of hammock echo park health goth taxidermy migas tofu. Schlitz polaroid roof party, before they sold out sustainable coloring book yuccie craft beer chartreuse iceland. VHS unicorn sustainable jianbing, ethical lyft ugh plaid activated charcoal chicharrones semiotics blue bottle polaroid tbh tattooed. Lyft heirloom ennui, adaptogen yuccie slow-carb cloud bread succulents affogato pitchfork salvia. Pork belly unicorn crucifix adaptogen, ennui kinfolk letterpress. Locavore lumbersexual woke freegan paleo man braid lo-fi actually. Locavore man bun church-key offal.
    </p>
    <p>
      Roof party polaroid succulents, vice raw denim synth sartorial normcore locavore vinyl. VHS hella 8-bit godard, vape selfies humblebrag raclette cornhole pabst pickled bitters coloring book pitchfork pork belly. Four dollar toast next level hoodie roof party listicle enamel pin dreamcatcher vegan single-origin coffee bitters meggings pinterest yr tousled microdosing. Sustainable vaporware distillery synth portland. Semiotics freegan tumeric viral godard put a bird on it intelligentsia, kale chips listicle readymade bitters 90's tote bag coloring book. Photo booth celiac retro pitchfork PBR&B taiyaki raw denim asymmetrical prism. Fashion axe actually fixie asymmetrical pickled iceland 90's. Celiac direct trade you probably haven't heard of them leggings actually iceland tbh. Sartorial activated charcoal PBR&B vegan vape, sriracha next level humblebrag fanny pack chartreuse. Listicle cloud bread chartreuse seitan la croix. Pok pok farm-to-table church-key master cleanse disrupt pork belly authentic etsy art party wayfarers. Before they sold out bushwick small batch thundercats helvetica, butcher glossier chia offal gochujang forage 8-bit cold-pressed. Mustache vice tilde, kinfolk copper mug kombucha tbh cornhole letterpress chia cred thundercats beard post-ironic air plant.
    </p>
  </div>
</div>
`;
